src/variables.rs

```diff
@@ -55,16 +55,45 @@ async fn input_item(name: &str, mut item: &Descriptor, all: &InputTypedesc,
     }
     match item {
         Descriptor::BaseScalar(s) => {
-            if s.id == codec::STD_STR {
-                let val = match state.variable_input(name, "str", "").await {
-                    | prompt::Input::Text(val) => val,
-                    | prompt::Input::Interrupt
-                    | prompt::Input::Eof => Err(Canceled)?,
-                };
-                Ok(Value::Str(val))
-            } else {
-                Err(anyhow::anyhow!(
-                    "Unimplemented input type {}", s.id))
+            let type_name = match s.id {
+                codec::STD_STR => "str",
+                codec::STD_INT16 => "int16",
+                codec::STD_INT32 => "int32",
+                codec::STD_INT64 => "int64",
+                _ => return Err(anyhow::anyhow!(
+                        "Unimplemented input type {}", s.id))
+            };
+
+            let val = match state.variable_input(name, type_name, "").await {
+                | prompt::Input::Text(val) => val,
+                | prompt::Input::Interrupt
+                | prompt::Input::Eof => Err(Canceled)?,
+            };
+
+            match s.id {
+                codec::STD_STR => {
+                    Ok(Value::Str(val))
+                }
+                codec::STD_INT16 => {
+                    match val.parse::<i16>() {
+                        Ok(v) => Ok(Value::Int16(v)),
+                        Err(_) => Err(anyhow::anyhow!("invalid int16 value"))
```

Comment:
This is not helpful enough, rust is able to show more specific error. It should work by providing context:
```use anyhow::Context;
val.parse::<i16>().context("invalid int16 value")?;
```

Reply:
I've implemented the suggestion, however the error output doesn't necessarily look better now:

```
yury> select <int16>$a;
Parameter <int16>$a: 1231231231
Error {
    context: "invalid int16 value",
    source: ParseIntError {
        kind: Overflow,
    },
}
```

Is that what you expected?

Reply:
The error is fine. The following should fix error formatting:
```diff
diff --git a/src/client.rs b/src/client.rs
index 80c2acd..64967d3 100644
--- a/src/client.rs
+++ b/src/client.rs
@@ -521,7 +521,7 @@ async fn _interactive_main(
             let input = match input_variables(&indesc, state).await {
                 Ok(input) => input,
                 Err(e) => {
-                    eprintln!("{:#?}", e);
+                    eprintln!("{:#}", e);
                     state.last_error = Some(e);
                     continue 'statement_loop;
                 }
```
Also we might want to loop inside this function until the value is valid or Ctrl+C. But this can be in another commit.

Reply:
Nice! The error output is definitely better now. I'll merge the PR.

> Also we might want to loop inside this function until the value is valid or Ctrl+C. But this can be in another commit.

I agree. Let's first decide on https://github.com/edgedb/edgedb-cli/issues/15.

Thanks for the review